fe/fe-core/src/main/java/com/starrocks/http/rest/transaction/TransactionWithChannelHandler.java

```diff
@@ -58,8 +59,11 @@ public ResultWrapper handle(BaseRequest request, BaseResponse response) throws U
                             "Channel ID should be between [0, %d].", (channel.getNum() - 1)));
                 }
 
+                String warehouseName = txnOperationParams.getWarehouseName();
```

Comment:
is it possible the warehouseName is not set in the params, or the warehouseName doesn't exist in the following getWarehouse() call.

This part of code doesn't check current runMode, but always check the warehouse exists, may introduce unexpected NPE in shared-nothing mode?

Reply:
thanks for this reminding, to make things clear, i have refactored the code to considering run mode